cdap-client/src/main/java/io/cdap/cdap/client/DatasetClient.java

```diff
@@ -240,7 +240,10 @@ public void truncate(DatasetId instance)
     throws IOException, UnauthenticatedException, UnauthorizedException {
     URL url = config.resolveNamespacedURLV3(instance.getParent(),
                                             String.format("data/datasets/%s/admin/truncate", instance.getDataset()));
-    restClient.execute(HttpMethod.POST, url, config.getAccessToken());
+    HttpRequest.Builder request = HttpRequest.post(url);
+    //Important to add body even if runtimeArgs is null to avoid 411 error for Http POST
+    request.withBody("");
```

Comment:
You can combine the `withBody` and `build` to
```
HttpRequest request = HttpRequest.post(url).withBody("").build();
```

Reply:
Done